src/java.base/unix/classes/java/lang/ProcessEnvironment.java

```diff
@@ -52,37 +52,96 @@
  * @since   1.5
  */
 
+/*
+ * ===========================================================================
+ * (c) Copyright IBM Corp. 2022, 2022 All Rights Reserved
+ * ===========================================================================
+ */
+
 package java.lang;
 
 import java.io.*;
 import java.util.*;
-
+import openj9.internal.criu.InternalCRIUSupport;
 
 final class ProcessEnvironment
 {
     private static final HashMap<Variable,Value> theEnvironment;
     private static final Map<String,String> theUnmodifiableEnvironment;
+    // CRIU enable flag
+    private static final boolean isCRIUEnabled;
+    // 1 - prints a message if the env var was set but not in the immutable list
+    // 2 - throws an exception
+    private static final int tracePrunedEnvVarsValue;
+    private static final Map<String,String> theOriginalUnmodifiableEnvironment;
     static final int MIN_NAME_LENGTH = 0;
 
     static {
+        isCRIUEnabled = InternalCRIUSupport.isCRIUSupportEnabled();
```

Comment:
also we can only take this path if the criu property is set, so that needs to be checked first before everything else

Reply:
Fixed via `CRIUProvider.getOpenJ9CRIU()`.